examples/dynamic-bridge-app/linux/main.cpp

```diff
@@ -176,6 +205,127 @@ std::unique_ptr<GeneratedCluster> CreateCluster(chip::ClusterId id)
     return nullptr;
 }
 
+CHIP_ERROR ReadValueFromBuffer(chip::TLV::TLVWriter & wr, const Span<const char> & v)
+{
+    return wr.PutString(chip::TLV::Tag(), v);
+}
+CHIP_ERROR ReadValueFromBuffer(chip::TLV::TLVWriter & wr, const bool & v)
+{
+    return wr.PutBoolean(chip::TLV::Tag(), v);
+}
+
+template <typename T>
+CHIP_ERROR ReadValueFromBuffer(chip::TLV::TLVWriter & wr, const T & v)
+{
+    return wr.Put(chip::TLV::Tag(), v);
+}
+
+CHIP_ERROR WriteValueToBuffer(const bool & value, uint8_t * buffer, const uint16_t & maxReadLength)
+{
+    if (maxReadLength != 1)
+    {
+        return CHIP_ERROR_BUFFER_TOO_SMALL;
+    }
+    *buffer = value ? 1 : 0;
+    return CHIP_NO_ERROR;
+}
+
+template <typename T>
+CHIP_ERROR WriteValueToBuffer(const T & value, uint8_t * buffer, const uint16_t & maxReadLength)
+{
+    size_t value_size = sizeof(value);
+    if (maxReadLength < value_size)
+    {
+        return CHIP_ERROR_BUFFER_TOO_SMALL;
+    }
+    memcpy(buffer, &value, value_size);
```

Comment:
if sizeof(value) is smaller than the max length, should the caller not actually know how much was written? Why is that not important? Explain in a comment if it is not.

Reply:
These functions follow a similar approach used in bridge-app (see `HandleReadOnOffAttribute` in `examples/bridge-app/linux/main.cpp`), where it expects exactly `maxReadLength` bytes to be written into the buffer, giving an error otherwise. Furthermore, the emberAf internal caller function does not expect to receive number of bytes written either (see `emAfReadOrWriteAttribute`).